File: frontends/main/src/app/configurator/services/api/workflow-action.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {WorkflowAction} from '@core/model/workflow-action';

@Injectable({
	providedIn: 'root'
})
export class WorkflowActionService {
	constructor(private http: HttpClient) {}

	getWorkflowActions(projectId: string): Observable<WorkflowAction[]> {
		return this.http.get<WorkflowAction[]>(`/api/superuser/configurator/projects/${projectId}/config/workflow-actions`);
	}

	getWorkflowAction(projectId: string, workflowActionId: string): Observable<WorkflowAction> {
		return this.http.get<WorkflowAction>(`/api/superuser/configurator/projects/${projectId}/config/workflow-actions/${workflowActionId}`);
	}

	createWorkflowAction(projectId: string, workflowAction: WorkflowAction): Observable<WorkflowAction> {
		return this.http.post<WorkflowAction>(`/api/superuser/configurator/projects/${projectId}/config/workflow-actions`, workflowAction);
	}

	updateWorkflowAction(projectId: string, workflowActionId: string, workflowAction: WorkflowAction): Observable<WorkflowAction> {
		return this.http.put<WorkflowAction>(`/api/superuser/configurator/projects/${projectId}/config/workflow-actions/${workflowActionId}`, workflowAction);
	}

	deleteWorkflowAction(projectId: string, workflowActionId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/workflow-actions/${workflowActionId}`);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/WorkflowActionService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.workflow.WorkflowActionDTO;

public interface WorkflowActionService {

	List<WorkflowActionDTO> getWorkflowActions(UUID projectId);

	WorkflowActionDTO getWorkflowAction(UUID projectId, UUID workflowActionId);

	WorkflowActionDTO createWorkflowAction(UUID projectId, WorkflowActionDTO workflowAction);

	WorkflowActionDTO updateWorkflowAction(UUID projectId, UUID workflowActionId, WorkflowActionDTO workflowAction);

	void deleteWorkflowAction(UUID projectId, UUID workflowActionId);
}
